packages/pipeline-editor/src/PipelineService.tsx

```diff
@@ -83,28 +92,49 @@ export class PipelineService {
       true
     );
 
-    const dialogTitle = 'Job submission to ' + runtimeName + ' succeeded';
-    const dialogBody = (
-      <p>
-        Check the status of your pipeline at{' '}
-        <a href={response['run_url']} target="_blank" rel="noopener noreferrer">
-          Run Details.
-        </a>
-        <br />
-        The results and outputs are in the {
-          response['object_storage_path']
-        }{' '}
-        working directory in{' '}
-        <a
-          href={response['object_storage_url']}
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          object storage
-        </a>
-        .
-      </p>
-    );
+    let dialogTitle;
+    let dialogBody;
+    if (response['run_url']) {
+      // pipeline executed remotely in a runtime of choice
+      dialogTitle = 'Pipeline submission to ' + runtimeName + ' succeeded';
+      dialogBody = (
+        <p>
+          Check the status of your pipeline at{' '}
+          <a
+            href={response['run_url']}
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            Run Details.
+          </a>
+          <br />
+          The results and outputs are in the {
+            response['object_storage_path']
+          }{' '}
+          working directory in{' '}
+          <a
+            href={response['object_storage_url']}
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            object storage
+          </a>
+          .
+        </p>
+      );
+    } else {
+      // pipeline executed in-place locally
+      dialogTitle = 'Pipeline execution in-place locally succeeded';
```

Comment:
The temporary directory will contain failure information as well.  How does a user determine what went wrong with local execution?

Reply:
Papermill updates the notebook, with failure details. The location of the execution logs is also displayed to the user on the response dialog box in case they need to find more info or investigate any issues.

Reply:
Sorry, I missed the fact that `_execute_file()` eats all errors and thought an exception would bypass the creation of the `PipelineProcessorResponse()` instance - like it does for the other runtime.

Reply:
The text here feels off, might be an issue with word ordering.